flink-streaming-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraTupleWriteAheadSink.java

```diff
@@ -110,11 +92,25 @@ public void close() throws Exception {
 	}
 
 	@Override
-	protected void sendValues(Iterable<IN> values, long timestamp) throws Exception {
-		//verify that no query failed until now
-		if (exception != null) {
-			throw new Exception(exception);
-		}
+	protected boolean sendValues(Iterable<IN> values, long timestamp) throws Exception {
+		int updatesSent = 0;
+		final AtomicInteger updatesConfirmed = new AtomicInteger(0);
+
+		final AtomicContainer<Throwable> exception = new AtomicContainer<>();
+
+		FutureCallback<ResultSet> callback = new FutureCallback<ResultSet>() {
+			@Override
+			public void onSuccess(ResultSet resultSet) {
+				updatesConfirmed.incrementAndGet();
+			}
+
+			@Override
+			public void onFailure(Throwable throwable) {
+				exception.set(throwable);
```

Comment:
i don't think it matters too much, as long as _some_ exception is noticed. The first exception would probably be the most reasonable approach though.